src/resolver/strategies/openapi-3-1-apidom/index.js

```diff
@@ -1,14 +1,28 @@
+import { isPlainObject } from 'ramda-adjunct';
+import { isElement } from '@swagger-api/apidom-core';
+
 import resolveOpenAPI31Strategy from './resolve.js';
 import normalize, { pojoAdapter } from './normalize.js';
 import { isOpenAPI31 } from '../../../helpers/openapi-predicates.js';
 
 const openApi31ApiDOMStrategy = {
   name: 'openapi-3-1-apidom',
-  match({ spec }) {
+  match(spec) {
     return isOpenAPI31(spec);
   },
-  normalize({ spec }) {
-    return pojoAdapter(normalize)(spec);
+  normalize(spec) {
+    // pre-normalization - happens only once before the first lazy dereferencing and in JavaScript context
+    if (!isElement(spec) && isPlainObject(spec) && !spec.$$normalized) {
```

Comment:
Didn't we remove setting the `$$normalized` field so `!spec.$$normalized` is unnecessary here?